src/main/java/com/javadeobfuscator/deobfuscator/transformers/stringer/HideAccessObfuscationTransformer.java

```diff
@@ -448,4 +686,82 @@ private boolean isEqualType(String type1, String type2) {
         }
         return false;
     }
+    
+    private String getPrimitiveFromClass(String clazz) {
+        String[][] types = {
+                {"java/lang/Byte", "byte"},
+                {"java/lang/Short", "short"},
+                {"java/lang/Integer", "int"},
+                {"java/lang/Long", "long"},
+                {"java/lang/Float", "float"},
+                {"java/lang/Double", "double"},
+                {"java/lang/Character", "char"},
+                {"java/lang/Boolean", "boolean"}
+        };
+
+        for (String[] type : types) {
+        	if(clazz.equals(type[0]))
+        		return type[1];
+        }
+        return null;
+    }
+    
+    private String convertPrimitiveToClassArray(String array, int mode)
+    {
+    	if(mode == 0)
+    	{
+	    	String[][] types = {
+	    		{"byte[]", "java/lang/Byte[]"},
+	    		{"short[]", "java/lang/Short[]"},
+	    		{"int[]", "java/lang/Integer[]"},
+	    		{"long[]", "java/lang/Long[]"},
+	    		{"float[]", "java/lang/Float[]"},
+	    		{"double[]", "java/lang/Double[]"},
+	    		{"char[]", "java/lang/Character[]"},
+	    		{"boolean[]", "java/lang/Boolean[]"}
+	    	};
+	
+	    	for (String[] type : types) {
+	     	if(array.equals(type[0]))
+	     		return type[1];
+	    	}
+	    	return array;
+    	}else if(mode == 1)
+    	{
+    		String[][] types = {
+	    		{"[B", "Ljava/lang/Byte[]"},
+	    		{"[S", "Ljava/lang/Short[]"},
+	    		{"[I", "Ljava/lang/Integer[]"},
+	    		{"[L", "Ljava/lang/Long[]"},
+	    		{"[F", "Ljava/lang/Float[]"},
+	    		{"[D", "Ljava/lang/Double[]"},
+	    		{"[C", "Ljava/lang/Character[]"},
+	    		{"[Z", "Ljava/lang/Boolean[]"}
+	    	};
+	
+	    	for (String[] type : types) {
+	     	if(array.equals(type[0]))
+	     		return type[1];
+	    	}
+	    	return array;
+    	}
+    	return array;
+    }
+    
+    /**
+     * Determines if class1 is implemented at any point by class2.
+     * This means that casting class2 to class1 would be casting from
+     * more specific to less specific.
+     * @param class1 Should have slashes instead of dots
+     */
+    private boolean isInterfaceOf(String class1, JavaClass class2)
```

Comment:
Can we use `Deobfuscator#isSubclass`?

Reply:
I'm not sure how that's loaded, but it needs to handle multiple "implements".
Which means that if a implements b and b implements c, isInterfaceOf(a, c) should return true.

Reply:
Should work fine. It's what overrides ASM's default subclass check for it to calculate frames